sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -50,161 +50,162 @@
 #define TIMEOUT_MQTT_RECEIVE_MS (60 * 1000)
 #define TIMEOUT_MQTT_DISCONNECT_MS (10 * 1000)
 
-static char mqtt_client_id[128];
-static char mqtt_username[128];
-static char register_publish_topic[128];
-static char query_topic[256];
-static char global_provisioning_endpoint[256] = { 0 };
-static char id_scope[16] = { 0 };
-static char registration_id[256] = { 0 };
-static char x509_cert_pem_file[256] = { 0 };
-static char x509_trust_pem_file[256] = { 0 };
+// Logging
+#define LOG_ERROR(...) \
+  { \
+    (void)printf("ERROR: %s:%s:%d: ", __FILE__, __func__, __LINE__); \
+    (void)printf(__VA_ARGS__); \
+    fflush(stdout); \
+    fflush(stderr); \
+  }
+#define LOG_SUCCESS(...) \
+  { \
+    (void)printf("SUCCESS: "); \
+    (void)printf(__VA_ARGS__); \
+  }
 
+// Buffers
+static char x509_cert_pem_file_buffer[256] = { 0 };
+static char x509_trust_pem_file_buffer[256] = { 0 };
+static char global_provisioning_endpoint_buffer[256] = { 0 };
+static char id_scope_buffer[16] = { 0 };
+static char registration_id_buffer[256] = { 0 };
+static char mqtt_client_id_buffer[128];
```

Comment:
nit: Why are some initialized but others aren't? Maybe add a comment.

Reply:
I'm not sure we should be doing it for any of them. Statically allocated buffers are guaranteed to be 0'd anyways.

Reply:
@ahsonkhan thanks for pointing out the inconsistency. @danewalton you're right about initialization.  I think I'll just remove all of the  ={0};